### PR TITLE
fix(backend): support intra-rafiki payments

### LINKED ISSUE #174: Unable to send packets to self
`createOutgoingPayments` is incorrectly passing a payment pointer to `makeIlpPlugin`, which needs to be able to look up a source account.
https://github.com/interledger/rafiki/blob/main/packages/backend/src/outgoing_payment/service.ts#L90
Further, `handleIlpData` didn't rename `ctx.state.account` to `incomingAccount`
https://github.com/interledger/rafiki/blob/main/packages/backend/src/connector/core/rafiki.ts#L165

We _could_ create the outgoing payment's account before payment setup, but we'd then have created unusable tigerbeetle balances whenever payment setup fails.

Instead, the sending payment pointer's asset can have an account used for payment setup.
If we convert assets' liquidity balance to an actual account, we'd have the added benefit of the liquidity service only having do deal with accounts (instead of assets and accounts)
https://github.com/interledger/rafiki/blob/main/packages/backend/src/liquidity/service.ts#L16-L20

### LINKED ISSUE #179: Fix debit vs credit
Debit and credit accounts are swapped throughout Rafiki :upside_down_face: 
https://github.com/interledger/rafiki/blob/main/packages/backend/src/transfer/service.ts#L86-L87
Accounts/balances should default to `debitBalance`
https://github.com/interledger/rafiki/blob/main/packages/backend/src/balance/service.ts#L11

### PR DESCRIPTION
**Pull request by wilsonianb**
## Changes proposed in this pull request

- Fix ilp-plugin and connector middleware to support intra-rafiki outgoing payments to invoices.
- Change asset balances to accounts
- Make credit/debit balance type explicit

## Context

- fixes #174 
- fixes #179 

## Checklist

- [x] Related issues linked using `fixes #number`
- [x] Tests added/updated
- [ ] Documentation added
- [x] Make sure that all checks pass

### CODE REVIEW

**Comment by matdehaast** [comment]
@wilsonianb can we add in the README/Docs just a reference on the meaning of a DEBIT and CREDIT in Rafiki context

**Review by matdehaast — DISMISSED** [review]
This looks good @wilsonianb .

I just have one question. If there is a 1-1 mapping between an Asset and AssetAccounts, what is the purpose of having two tables for this?

**Comment by wilsonianb** [comment]
Adding the account ids foreign keys to the `assets` table in a subsequent migration does work, but inserting an asset row gets complicated because the account ids should be `notNullable` but you need to know the asset's smallserial `unit` value before the accounts can be created, which isn't known until postgres assigns it when the asset is inserted.

File: packages/backend/src/account/model.ts [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-import { Model } from 'objection'
+import { Model, RelationMappings } from 'objection'
 
 export class Account extends BaseModel {
   public static get tableName(): string {
     return 'accounts'
   }
 
-  static relationMappings = {
+  static relationMappings = (): RelationMappings => ({
```
[the remark this answers is not in the document]
Comment (wilsonianb) [reply packages/backend/src/account/model.ts:10]: It avoids circular dependency issues between `Account` and `Asset`
https://vincit.github.io/objection.js/guide/relations.html#require-loops
https://github.com/Vincit/objection.js/blob/216e897738db4aa10039039422fa590ca9cc59cb/tests/ts/examples.ts#L328

**Comment by wilsonianb** [comment]
The failing checks are due to this tigerbeetle-node issue
https://github.com/coilhq/tigerbeetle-node/issues/20

File: packages/backend/src/balance/service.ts [hunk 2]
```diff
@@ -7,12 +7,17 @@ import { uuidToBigInt } from '../shared/utils'
 
 const ACCOUNT_RESERVED = Buffer.alloc(48)
 
+export enum BalanceType {
+  Credit = 'Credit', // >= 0
+  Debit = 'Debit' // <= 0
```
[the remark this answers is not in the document]
Comment (wilsonianb) [reply packages/backend/src/balance/service.ts]: updated https://github.com/interledger/rafiki/pull/182/commits/1cc5adee7218f6ac8465ecc6745b8027567bdbed

**Comment by wilsonianb** [comment]
I'm going to go ahead and merge.
Some added comments are the only change since checks passed for https://github.com/interledger/rafiki/pull/182/commits/74d7100726f859d7995fefa0aaca5b45c7c11b22.
`yarn install --immutable --immutable-cache` will fail with a fresh checkout of `main` due to the tigerbeetle-node issue with or without these changes.

**Pull request merged**